codex-cli/src/shims-external.d.ts

```diff
@@ -0,0 +1,24 @@
+// Ambient module declarations for optional/runtime‑only dependencies so that
+// `tsc --noEmit` succeeds without installing their full type definitions.
+
+declare module "package-manager-detector" {
+  export type AgentName = "npm" | "pnpm" | "yarn" | "bun" | "deno";
+
+  /** Detects the package manager based on environment variables. */
+  export function getUserAgent(): AgentName | null | undefined;
+}
+
+declare module "fast-npm-meta" {
+  export interface LatestVersionMeta {
+    version: string;
+  }
+
+  export function getLatestVersion(
+    pkgName: string,
+    opts?: Record<string, unknown>,
+  ): Promise<LatestVersionMeta | { error: unknown }>;
+}
+
+declare module "semver" {
```

Comment:
https://www.npmjs.com/package/@types/semver exists but not the others, apparently?